cql3/query_processor.cc

```diff
@@ -1025,16 +1026,30 @@ query_processor::execute_schema_statement(const statements::schema_altering_stat
 
     cql3::cql_warnings_vec warnings;
 
+    auto request_id = guard->new_group0_state_id();
+    // TODO: either save as a member of statements::schema_altering_statement class
+    //       or extend alter_keyspace_statement::prepare_schema_mutations with guard's req id
+    const_cast<statements::schema_altering_statement&>(stmt).global_req_id = fmt::to_string(request_id);
+
     auto [ret, m, cql_warnings] = co_await stmt.prepare_schema_mutations(*this, options, guard->write_timestamp());
     warnings = std::move(cql_warnings);
 
+    ce = std::move(ret);
     if (!m.empty()) {
         auto description = format("CQL DDL statement: \"{}\"", stmt.raw_cql_statement);
-        co_await remote_.get().mm.announce(std::move(m), std::move(*guard), description);
+        if (ce && ce->target == cql_transport::event::schema_change::target_type::TABLET_KEYSPACE) {
+            co_await remote_.get().mm.announce<service::topology_change>(std::move(m), std::move(*guard), description);
+            // TODO: eliminate timeout from alter ks statement
```

Comment:
Should `wait_for_topology_request_completion` receive a timeout option?

Reply:
It refers to cqlsh/drivers timeout and these 2 comments under RFC - https://github.com/scylladb/scylladb/issues/16129#issuecomment-2026939776
What do you want me to do with this TODO comment? Just make it more specific and link these 2 comments is enough you think?

Reply:
I'll leave this comment, as driver change hasn't been performed, to the best of my knowledge.